### PR TITLE
Assets: Load files directly for UNPACKED_MPQS

### PR DESCRIPTION
**Pull request by glebm**
When using `UNPACKED_MPQS`, avoid all the SDL rwops machinery and use `<cstdio>` / `file_utils` instead.

Benefits:
1. Reduced indirection.
2. Can test for the file's existence and get the file size without opening it, which is much faster.

This also cleans up file handling a bit:
1. Eliminates the need for manual handle management.
3. Separates finding files from opening them.

### CODE REVIEW

File: Source/engine/assets.cpp [hunk 1]
```diff
@@ -65,50 +67,140 @@ bool OpenMpqFile(const char *filename, MpqArchive **archive, uint32_t *fileNumbe
...
+#endif
+
+	return result;
+}
 #endif
 
-	return nullptr;
+AssetHandle OpenAsset(AssetRef &&ref, bool threadsafe)
+{
+#if UNPACKED_MPQS
```
Comment (AJenbo) [inline Source/engine/assets.cpp:162]:
```suggestion
#ifdef UNPACKED_MPQS
```

File: Source/engine/assets.cpp [hunk 2]
```diff
@@ -65,50 +67,140 @@ bool OpenMpqFile(const char *filename, MpqArchive **archive, uint32_t *fileNumbe
...
+}
+
+SDL_RWops *OpenAssetAsSdlRwOps(const char *filename, bool threadsafe)
+{
+#ifdef UNPACKED_MPQS
+	AssetRef ref = FindAsset(filename);
+	if (!ref.ok())
+		return nullptr;
+	return SDL_RWFromFile(ref.path, "rb");
+#else
```
Comment (AJenbo) [inline Source/engine/assets.cpp:201]: Isn't the SDL path only for `#ifndef`?

**Review by AJenbo — CHANGES_REQUESTED** [review]
A couple of the ifdefs look suspect

File: Source/engine/assets.cpp (hunk 2 above)
Comment (glebm) [reply Source/engine/assets.cpp:201]: We still use the SDL path for video rendering and music playback, which both accept only SDL_RWops

File: Source/engine/assets.cpp (hunk 1 above)
Comment (glebm) [reply Source/engine/assets.cpp:162]: Thanks, fixed

**Review by AJenbo — APPROVED** [review]

**Pull request merged**